src/errors/ValidationError.ts

```diff
@@ -0,0 +1,16 @@
+// This error was donated by the TV Kitchen project / Bad Idea Factory
+// https://github.com/tvkitchen/countertop/blob/main/src/errors/ValidationError.ts
+import type { ErrorObject } from 'ajv';
+
+export class ValidationError extends Error {
+  public errors?: ErrorObject[] | null;
+
+  public constructor(
+    message?: string,
+    errors?: ErrorObject[] | null,
+  ) {
+    super(message);
+    this.name = this.constructor.name;
+    this.errors = errors;
+  }
+}
```

Comment:
This `ValidationError` class is great. For a second I wondered if `ErrorObject` also extends `Error` but no, it does not. My tendency would be to make as many fields mandatory and immutable only as possible for this commit and then relax them to optional as needed in future commits. Perhaps that already happened in the course of TV Kitchen, though, and so making them optional would be inevitable here anyway.

Reply:
That's a good thought and I agree (I forget if the TVK had cases where it needed to be optional but it's so easy to add it later if we find one)